### DISCUSSION TITLE
API routes works locally, but not in production

### DISCUSSION THREAD

**Discussion by kaf-lamed-beyt** [vercel/next.js#43776] [Help] [answered]
I have a `posts` collection in my database (I'm using MongoDB) where I update the views of the articles whenever someone visits the unique slug of each article.

The issue now here is that, when I hit the endpoint locally, on my PC, everything works fine. But, when I deploy it to netlify or vercel, I keep getting this error "Bad request, missing form".

What could be wrong? 

PS: I'm not using next13, yet. Current version is `^12.2.1`

**Comment by kaf-lamed-beyt** [comment]
@icyJoseph, any idea on what the issue could be here?

**Comment by icyJoseph** [comment]
Is the DB reachable when deployed? Where is it hosted?

**Reply by kaf-lamed-beyt** [reply]
I have no idea if it is or not.

I think the cluster is located around the Paris-EU region (If I can recall clearly)

**Comment by kaf-lamed-beyt** [comment, marked as answer]
I figured out what the issue is, and I wrote about it [here](https://meje.dev/blog/next-directory-returns-error-404-on-netlify). Kindly take a look.

**Reply by icyJoseph** [reply]
Ah, yes, `next export` produces a static HTML page, AKA no server.

https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features:

> [Unsupported Features](https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features)
> Features that require a Node.js server, or dynamic logic that cannot be computed during the build process, are not supported:
> 
> [Image Optimization](https://nextjs.org/docs/basic-features/image-optimization) (default loader)
> [Internationalized Routing](https://nextjs.org/docs/advanced-features/i18n-routing)
> [API Routes](https://nextjs.org/docs/api-routes/introduction)
> [Rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites)
> [Redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects)
> [Headers](https://nextjs.org/docs/api-reference/next.config.js/headers)
> [Middleware](https://nextjs.org/docs/middleware)
> [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)
> [fallback: true](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true)
> [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props)


Good that you figured it out! With this much info it was a hard guess.

**Reply by kaf-lamed-beyt** [reply]
Thank you for your help, nonetheless! 😎
